adagrams/game.py

```diff
@@ -1,11 +1,159 @@
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    import random
+    list_of_letters = list(LETTER_POOL.keys()) 
+    weight_letters = list(LETTER_POOL.values())
+    ten_letters = []
+    letter_freq = {}
+
+    # could create a helperfunction if the letters are more then 10
+    while len(ten_letters) < 10:
+        list_letter_drawn = random.choices(list_of_letters, weights = weight_letters, k=1)
+        letter_drawn = list_letter_drawn[0]
+        if letter_drawn in letter_freq:
+            if letter_freq[letter_drawn] < LETTER_POOL[letter_drawn]:
+                letter_freq[letter_drawn] +=1
+            else:
+                continue
+        else:
+            letter_freq[letter_drawn] = 1
+
+        ten_letters.append(letter_drawn)
+
+
+    return ten_letters        
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+#  """Has two parameters:
+#    - `word`, the first parameter, describes some input word, and is a string
+#    - `letter_bank`, the second parameter, describes an array of drawn letters in a hand. You can expect this to be an array of ten strings, with each string representing a letter
+#     - Returns either `True` or `False`
+#     - Returns `True` if every letter in the `input` word is available (in the right quantities) in the `letter_bank`
+#     - Returns `False` if not; if there is a letter in `input` that is not present in the `letter_bank` or has too much of compared to the `letter_bank`
+# """ 
+    letter_bank_count = {}
+
+    for letters in letter_bank:
+        if letters in letter_bank_count:
+            letter_bank_count[letters] +=1
+        else:
+            letter_bank_count[letters] = 1
+
+    word_capital = word.upper()
+    for letter in word_capital:
+        if letter in letter_bank and letter_bank_count[letter]>=1:
+            letter_bank_count[letter] -=1  
+        else:
+            return False
+    return True
```

Comment:
You could simplify this function by making a copy of the letter bank using `deepcopy`. Once we are iterating over the words we can add a conditional to find out if the letter in the word is in the letter bank copy. Something like this:
```
for letter in word.upper():
        if letter in letter_bank_copy:
            letter_bank_copy.remove(letter)
        else:
            return False
```